contracts/fuelchain/FuelChainConsensus.sol

```diff
@@ -62,27 +81,40 @@ contract FuelChainConsensus is Initializable, PausableUpgradeable, AccessControl
         _unpause();
     }
 
-    /// @notice Sets the PoA key
-    /// @param key Address of the PoA authority
-    function setAuthorityKey(address key) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        _authorityKey = key;
+    /// @notice Commits a block header.
+    /// @param blockHash The hash of a block
+    /// @param epoch The epoch number of the commit
+    function commit(bytes32 blockHash, uint256 epoch) external whenNotPaused onlyRole(COMMITTER_ROLE) {
+        uint256 slot = epoch % NUM_COMMIT_SLOTS;
+        Commit storage commitSlot = _commitSlots[slot];
+        commitSlot.blockHash = blockHash;
+        commitSlot.timestamp = uint32(block.timestamp);
+
+        emit CommitSubmitted(epoch, blockHash);
     }
 
     //////////////////////
     // Public Functions //
     //////////////////////
 
-    /// @notice Gets the currently set PoA key
-    /// @return authority key
-    function authorityKey() public view returns (address) {
-        return _authorityKey;
+    /// @notice Checks if a given block is finalized
+    /// @param blockHash The hash of the block to check
+    /// @param blockHeight The height of the block to check
+    /// @return true if the block is finalized
+    function finalized(bytes32 blockHash, uint256 blockHeight) external view whenNotPaused returns (bool) {
+        uint256 epoch = blockHeight / BLOCKS_PER_EPOCH;
+        Commit storage commitSlot = _commitSlots[epoch % NUM_COMMIT_SLOTS];
+        require(commitSlot.blockHash == blockHash, "Unknown block");
+
+        return block.timestamp >= uint256(commitSlot.timestamp) + TIME_TO_FINALIZE;
     }
 
-    /// @notice Verify a given block.
-    /// @param blockHash The hash of a block
-    /// @param signature The signature over the block hash
-    function verifyBlock(bytes32 blockHash, bytes calldata signature) external view whenNotPaused returns (bool) {
-        return CryptographyLib.addressFromSignature(signature, blockHash) == _authorityKey;
+    /// @notice Gets the block hash at the given epoch
+    /// @param epoch The epoch number of the commit
+    /// @return hash of the block at the given epoch
+    function blockHashAtEpoch(uint256 epoch) external view returns (bytes32) {
+        Commit storage commitSlot = _commitSlots[epoch % NUM_COMMIT_SLOTS];
```

Comment:
Just want to make sure I understand the assignment of commit slots.
With the current config, we're basically  saying that for a given epoch, all commits within a 30-day window are stored in the same `_commitSlot`.

Reply:
There is a rolling buffer of commits we keep in storage which we target to last around 30 days before data gets overridden. The index that a particular block will sit in the buffer is deterministic (blockNumber / blocksPerEpoch) % bufferSize. Eventually there will be an enforced rule that says only the nth block in an epoch can be committed, but we haven't determined what that block will be yet (probably either the first or last). 

Side note: the term "epoch" is going to be changed to "commit interval". "Epoch" is now only going to refer to the amount of time in which the sequencer set stays constant before updating. This gives us more independent control over these separate parameters.